src/protocol/handshake.rs

```diff
@@ -152,6 +152,11 @@ impl Handshake for InnerNode {
                 rsp.extend_from_slice(
                     format!("X-Algorand-Genesis: {}\r\n", X_AG_ALGORAND_GENESIS).as_bytes(),
                 );
+                if let Some(challenge) = &self.challenge {
```

Comment:
How about `self.challenge.and_then` ?

Reply:
maybe `pub fn is_some_and(&self, f: impl FnOnce(&T) -> bool) -> bool` would be more applicable here, but we'll keep `if let`, it's simple enough